data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java

```diff
@@ -181,11 +193,42 @@ protected CircuitBreakerOptions createCircuitBreakerOptions(
 
   private HttpConsumerRecordSender createSender(
     final String target,
-    final CircuitBreakerOptions circuitBreakerOptions) {
+    final CircuitBreakerOptions circuitBreakerOptions,
+    final EgressConfig egress) {
+
+    final var circuitBreaker = CircuitBreaker.create(target, vertx, circuitBreakerOptions);
+
+    setRetryPolicy(egress, circuitBreaker);
 
     return new HttpConsumerRecordSender(
       client,
-      target
+      target,
+      circuitBreaker
     );
   }
+
+  private void setRetryPolicy(EgressConfig egress, CircuitBreaker circuitBreaker) {
```

Comment:
More than passing the `circuitBreaker` to configure it, can you return the `Function`?

Reply:
Also maybe name the method as `computeRetryPolicy`

Reply:
I like it!